contracts/ism-multisig/src/contract.rs

```diff
@@ -192,17 +75,35 @@ pub fn query(deps: Deps, _env: Env, msg: ISMQueryMsg) -> Result<Binary, Contract
 
     match msg {
         ModuleType => Ok(to_binary(&ISMType::Owned)?),
+        // TODO: ask what is stand for?
+        Verify {
+            metadata: raw_metadata,
+            message: raw_message,
+        } => {
+            let metadata: MessageIdMultisigIsmMetadata = raw_metadata.into();
+            let message: Message = raw_message.into();
+
+            let threshold = THRESHOLD.load(deps.storage, message.origin_domain.into())?;
+            let validators = VALIDATORS.load(deps.storage, message.origin_domain.into())?;
+
+            let mut signatures: Vec<Binary> = Vec::new();
+            for i in 0..metadata.signatures_len().unwrap() {
+                signatures.push(metadata.signature_at(i))
+            }
 
-        Verify { metadata, message } => {
-            let config = CONFIG.load(deps.storage)?;
-
-            let digest = sha256_digest(Binary::from(message))?;
+            let mut success: u8 = 0;
+            for validator in validators.0 {
```

Comment:
How about compare sorted validators pubkey and sorted signatures pubkey? https://github.com/CosmWasm/cosmwasm/blob/6a5c3bcf77ea4c789f79ab86efa668bfb5e020e7/packages/crypto/src/secp256k1.rs#L101 with `secp256k1_recover_pubkey`, you can derive pubkey from signature. - for recovery param, this only should be 0 or 1, so i believe the complexity of this method is better than iterating two vectors